src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER = 100;
+    private ColorSupplier randomColor = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        ColorSupplier randomColor = new ColorSupplier();
+        Random randFigure = new Random();
+
+        switch (randFigure.nextInt(5)) {
+            case 0:
+                Square square = new Square(randFigure.nextInt(1, NUMBER));
```

Comment:
```suggestion
                Figure square = new Square(randFigure.nextInt(1, NUMBER));
```
it is better to use parent variable type